### PR TITLE
wayvnc: update to 0.3.1.

### PR DESCRIPTION
**Pull request by maybe-one-day-ubermensch**

### CODE REVIEW

**Comment by ericonr** [comment]
@zdykstra

**Comment by zdykstra** [comment]
@ericonr built locally on x86_64-glibc, tested successfully under Sway with TigerVNC.

File: common/shlibs [hunk 1]
```diff
@@ -3987,7 +3987,7 @@ libocl.so.2019.07 opencamlib-2019.07_1
 libcglm.so.0 cglm-0.7.6_1
 libfcft.so.3 fcft-2.2.2_1
 libaml.so.0 aml-0.1.0_1
-libneatvnc.so.0 neatvnc-0.2.0_1
+libneatvnc.so.0 neatvnc-0.3.2_1
```
Comment (ericonr) [inline common/shlibs:3990]: This change isn't necessary, and, if it were, shouldn't be in this PR / in this commit. Otherwise, good to merge.

Comment (ericonr) [reply common/shlibs:3990]: This was probably a git accident, should be easily solvable.

**Comment by maybe-one-day-ubermensch** [comment]
At first I thought the shlib edit was unessecary, but after re-reading `wayvnc`'s `meson.build`, I saw `neatvnc_version = '>=0.3.0'`. The earliest version we have of `neatvnc` that matches that comparison is 3.2. In retrospect, I should've split the commit up into two and renamed the pr to updating `common/shlibs`.

**Comment by maybe-one-day-ubermensch** [comment]
`wayvnc` is also the only one that depends on `neatvnc`.

**Comment by ericonr** [comment]
Then `neatvnc` should ship a versioned shlib. I'm not sure what the best way of solving this conflict is, since adding a `depends to the template is even uglier. Can you check if it works with older `neatvnc`?

**Comment by maybe-one-day-ubermensch** [comment]
`wayvnc-3.1` doesn't compile with `neatvnc-2.0`.

**Comment by q66** [comment]
the changes are good as they are

**Comment by maybe-one-day-ubermensch** [comment]
Sounds good, I'll rename the pr and split up the commits.

**Pull request merged**